app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -171,16 +246,32 @@ protected void populateList(Query query) {
                 String id = (String) extractMap.get("id");
                 String days = extractMap.get("days").toString();
                 boolean isPublic = (boolean) extractMap.get("public");
+                Long sortIndex = (long) extractMap.get("sortIndex");
 
                 // create a new habit with the snapshot data
-                Habit addHabit = new Habit(title, reason, date.toDate(), days, isPublic, id, new HabitEventList());
+                Habit addHabit = new Habit(title, reason, date.toDate(), days, isPublic, id, new HabitEventList(), sortIndex);
 
                 // add the habit to the local list
                 _habitList.addHabitLocal(addHabit);
             }
         }
     }
 
+    /**
+     * This function gets the largest sort index in the list plus one
+     * @return Returns the largest sort index plus one
+     */
+    public Long getLargestSortIndex() {
+        long maxSortIndex = 0;
+        // This iterates through the habits to update their sort indexes
+        for (Habit habit : _habitItemAdapter._snapshots){
+            if (habit.getSortIndex() > maxSortIndex) {
+                maxSortIndex = habit.getSortIndex();
+            }
+        }
+        return maxSortIndex + 1;
+    }
```

Comment:
Just a minor thing. This method is named `getLargestSortIndex`, but the value it returns is the **next** possible sort index. I would either rename this method or increment by one outside this method.